### PR TITLE
Ethminer new arguments & unified pool protocol selection

### PR DESCRIPTION
**Pull request by papampi**
New ethminer command line arguments

### CODE REVIEW

**Comment by papampi** [comment]
Anything more needed here?

**Comment by LuKePicci** [comment]
Two points:
- also other miners do protocol selection the same way, it is better to generalize this option also, such that any *_SCHEMA= setting will apply the same way, you can default it to "stratum+tcp" when undefined
- the protocol I can choose is pool-specific, if I switch between ethash pools supporting different protocols slecting protocol in ethminer generic options isn't viable, so it should be included in coin specific settings

**Comment by LuKePicci** [comment]
```
UPOOL="_POOL"
UPORT="_PORT"
UPROTO="_POOL_PROTOCOL"
…
xport=$COIN$UPORT
xproto=$COIN$UPROTO
…

if [[ ${!xproto} == "" ]]; then eval $xproto="stratum+tcp"; fi
```

**Comment by papampi** [comment]
Didn't saw your comment on using  "proto", definitely better 
Already made changes ... will change em again

**Comment by papampi** [comment]
Didnt saw last part of your comment either ...

File: 1bash.template [hunk 1]
```diff
@@ -608,8 +610,9 @@ plusCPU_WALLET_FORMAT="$WALLET_ADDRESS_FORMAT"
 ########################################################
 
 CLAYMORE_VERSION="latest"                        # Choose: latest or any version you add its folder.
-ETHMINER_OPTS=""                                 # Ethminer optional arguments. add "--api-port -3333" if you want web info
 CLAYMORE_OPTS=""                                 # Claymore optional arguments. EX: add "-esm 2" for miningpoolhub
+ETHMINER_OPTS=""                                 # Ethminer optional arguments. add "--api-port -3333" if you want web info
+ETHMINER_SCHEMA="stratum+tcp"                    # Use Ethminer new schema; stratum+ssl, stratum+tcp, stratum+tls, stratum+tls12, stratum1+ssl, stratum1+tcp, stratum1+tls, stratum1+tls12, stratum2+ssl, stratum2+tcp, stratum2+tls, stratum2+tls12
```
Comment (LuKePicci) [inline 1bash.template]: Ok, just drop this line and move the comment  in the notes of the coin section and this is ready.

**Comment by papampi** [comment]
I think we should do the same for bminer so they can use different protocols for different coins.

**Comment by papampi** [comment]
Should be done if I'm not missing anything.

**Review by LuKePicci — DISMISSED** [review]
Awesome.

**Comment by papampi** [comment]
Looks like all is good, made a 20 coins switch to test.
Will merge later if all goes well.

**Pull request merged**